test/src/test/java/org/corfudb/integration/LockIT.java

```diff
@@ -153,13 +153,18 @@ public void testSingleLockClientMultipleLocks() throws Exception {
             // Since this is the only client, ALL locks should've been acquired, verify, block until condition is met
             waitCondition = WaitConditionType.LOCK_ACQUIRED;
             for (int i=0; i<numLocks; i++) {
-                log.debug("***** Wait until lock " + i + " is acquired");
+                log.debug("***** Wait until lock {} is acquired", i);
                 blockUntilWaitCondition.acquire();
             }
 
             assertThat(lockAcquiredObservables.get(clientId).getValue()).isEqualTo(numLocks);
-            lockIds.forEach(lockId ->
-                assertThat(client.getLocks().get(lockId).getState().getType()).isEqualTo(LockStateType.HAS_LEASE));
+            lockIds.forEach(lockId -> {
+                    while(client.getLocks().get(lockId).getState().getType() != LockStateType.HAS_LEASE) {
+                        // Lock state might take longer to update as the listener is updated before the FSM transitions
+                        // to the new state
+                    }
```

Comment:
Shouldn't there be a sleep here?

Reply:
And a max timeout wait before the test fails so that it doesn't keep on spinning indefinitely.

Reply:
There could be a sleep. The thing is the transition happens in the line right after the listener is notified that the lock was acquired. So technically it shouldn't be spinning for too long. It's just the interleaving between threads...  

Also, regarding the timeout. Isn't it covered by the max time of each test? otherwise it times out.